src/ethereum/utils/ensure.py

```diff
@@ -12,17 +12,11 @@
 Functions that simplify checking assertions and raising exceptions.
 """
 
-from typing import Callable
-
-
-class EnsureError(Exception):
-    """
-    Represents a failed check.
-    """
+from typing import Callable, Union
 
 
 def ensure(
-    value: bool, exception_class: Callable[[], Exception] = EnsureError
+    value: bool, exception: Union[Callable[[], BaseException], BaseException]
```

Comment:
Well that was a fruitless rabbithole to explore today. Python really needs an `Intersection` type hint...

With that pointless waste of time out of the way, I still think it would be better to use `Callable`:

```python
def ensure(
    value: bool, exception: Union[Callable[[], BaseException], BaseException]
) -> None:
    # ...
    if isinstance(exception, BaseException):
        raise exception
    else:
        raise exception()
```

It doesn't _exactly_ capture what `raise` requires, but I think it accepts a strict superset of `raise`'s behaviour, and catches the case where the exception requires an argument.